src/aes.rs

```diff
@@ -243,37 +250,37 @@ fn test_contract2() {
 
 #[test]
 fn test_kat_block1() {
-    let msg = Bytes::from(vec![
+    let msg = Block(secret_bytes!([
         0x6b, 0xc1, 0xbe, 0xe2, 0x2e, 0x40, 0x9f, 0x96, 0xe9, 0x3d, 0x7e, 0x11, 0x73, 0x93, 0x17,
-        0x2a,
-    ]);
-    let key = Key::from([
-        0x2b, 0x7e, 0x15, 0x16, 0x28, 0xae, 0xd2, 0xa6, 0xab, 0xf7, 0x15, 0x88, 0x9, 0xcf, 0x4f,
-        0x3c,
+        0x2a
+    ]));
+    let key = Key(secret_bytes!([
+        0x2b, 0x7e, 0x15, 0x16, 0x28, 0xae, 0xd2, 0xa6, 0xab, 0xf7, 0x15, 0x88, 0x09, 0xcf, 0x4f,
+        0x3c
+    ]));
+    let ctxt = secret_bytes!([
+        0x3a, 0xd7, 0x7b, 0xb4, 0x0d, 0x7a, 0x36, 0x60, 0xa8, 0x9e, 0xca, 0xf3, 0x24, 0x66, 0xef,
+        0x97
     ]);
-    let ctxt = [
-        0x3a, 0xd7, 0x7b, 0xb4, 0xd, 0x7a, 0x36, 0x60, 0xa8, 0x9e, 0xca, 0xf3, 0x24, 0x66, 0xef,
-        0x97,
-    ];
 
-    let c = aes128_encrypt_block(key, Block::from(msg.raw()));
-    assert_eq!(ctxt[..], c[..]);
+    let c = aes128_encrypt_block(key, msg);
+    assert_bytes_eq!(ctxt, c);
 }
 
 #[test]
 fn test_kat_block2() {
-    let msg = Bytes::from(vec![
+    let msg = Block(secret_bytes!([
         0x53, 0x69, 0x6e, 0x67, 0x6c, 0x65, 0x20, 0x62, 0x6c, 0x6f, 0x63, 0x6b, 0x20, 0x6d, 0x73,
-        0x67,
-    ]);
-    let key = Key::from([
+        0x67
+    ]));
+    let key = Key(secret_bytes!([
         0xae, 0x68, 0x52, 0xf8, 0x12, 0x10, 0x67, 0xcc, 0x4b, 0xf7, 0xa5, 0x76, 0x55, 0x77, 0xf3,
-        0x9e,
-    ]);
-    let ctxt = Bytes::from(vec![
-        0x61, 0x5f, 0x9, 0xfb, 0x35, 0x3f, 0x61, 0x3b, 0xa2, 0x8f, 0xf3, 0xa3, 0xc, 0x64, 0x75,
-        0x2d,
-    ]);
-    let c = aes128_encrypt_block(key, Block::from(msg.raw()));
-    assert_eq!(ctxt.raw()[..], c[..]);
+        0x9e
+    ]));
+    let ctxt = Bytes::from_array(&secret_bytes!([
```

Comment:
All the `from_XXX` functions are workarounds for Rust not being able to infer types properly on `from`. But let me know if you have ideas how to make that work.

Reply:
Rust type inference does not work all the time, especially for `From` traits. I would prefer to rely on monophonic functions with explicit names in a systematic manner, since it is also easier for the formalization.